src/formatDate.js

```diff
@@ -9,6 +9,74 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  let newDate = '';
+  let fromCharacter = 0;
+  let month;
+  let day;
+  let year;
+  let yearFormat;
+
+  fromFormat.forEach((element) => {
+    const toCharacter = fromCharacter + element.length;
+
+    if (element === 'MM') {
+      month = date.slice(fromCharacter, toCharacter);
+      fromCharacter = fromCharacter + 3;
```

Comment:
The increment of 'fromCharacter' should be based on the length of the current 'element' and the separator. It's currently hardcoded to 3, which is incorrect if there's no separator or a separator of different length.